client/src/components/PortfolioTable.tsx

```diff
@@ -0,0 +1,741 @@
+import { t, Trans } from "@lingui/macro";
+import {
+  ColumnDef,
+  ColumnFiltersState,
+  FilterFn,
+  filterFns,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getFacetedMinMaxValues,
+  getFacetedRowModel,
+  getFacetedUniqueValues,
+  getFilteredRowModel,
+  getPaginationRowModel,
+  getSortedRowModel,
+  PaginationState,
+  Row,
+  Table,
+  useReactTable,
+} from "@tanstack/react-table";
+import _groupBy from "lodash/groupBy";
+import React, { Fragment } from "react";
+import { Link } from "react-router-dom";
+import { createRouteForFullBbl } from "routes";
+import { SupportedLocale } from "../i18n-base";
+import Helpers, { longDateOptions } from "../util/helpers";
+import { logAmplitudeEvent } from "./Amplitude";
+import { AddressRecord, HpdComplaintCount } from "./APIDataTypes";
+import { FilterContext, IFilterContext, MINMAX_DEFAULT } from "./PropertiesList";
+import "styles/PortfolioTable.scss";
+import { sortContactsByImportance } from "./DetailView";
+
+const FIRST_COLUMN_WIDTH = 130;
+export const MAX_TABLE_ROWS_PER_PAGE = 100;
+
+declare module "@tanstack/table-core" {
+  interface FilterFns {
+    arrIncludesSome: FilterFn<unknown>;
+    inNumberRange: FilterFn<unknown>;
+    isNonZero: FilterFn<unknown>;
+  }
+}
+
+const currencyFormater = new Intl.NumberFormat("en-us", {
+  currency: "USD",
+  style: "currency",
+  minimumFractionDigits: 0,
+  maximumFractionDigits: 0,
+});
+
+const isNonZero: FilterFn<any> = (row, columnId, value, addMeta) =>
+  value ? !!row.getValue(columnId) : true;
+
+type PortfolioTableProps = {
+  data: AddressRecord[];
+  headerTopSpacing: number | undefined;
+  i18n: I18n;
+  locale: SupportedLocale;
+  rsunitslatestyear: number;
+  getRowCanExpand: (row: Row<AddressRecord>) => boolean;
+};
+
+/**
+ * This component memoizes the portfolio table via React.memo
+ * in an attempt to improve performance, particularly on IE11.
+ */
+export const PortfolioTable = React.memo(
+  React.forwardRef<HTMLDivElement, PortfolioTableProps>((props, lastColumnRef) => {
+    const { data, i18n, locale, rsunitslatestyear, getRowCanExpand } = props;
+
+    const { filterContext, setFilterContext } = React.useContext(FilterContext);
+
+    const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>([]);
+
+    const [{ pageIndex, pageSize }, setPagination] = React.useState<PaginationState>({
+      pageIndex: 0,
+      pageSize: MAX_TABLE_ROWS_PER_PAGE,
+    });
+
+    const pagination = React.useMemo(
+      () => ({
+        pageIndex,
+        pageSize,
+      }),
+      [pageIndex, pageSize]
+    );
+
+    const columns = React.useMemo<ColumnDef<AddressRecord, any>[]>(
+      () => [
+        {
+          accessorFn: (row) => `${row.housenumber} ${row.streetname}`,
+          id: "address",
+          header: () => i18n._(t`Address`),
+          cell: ({ row }) => {
+            return (
+              <Link to={createRouteForFullBbl(row.original.bbl, locale)}>
```

Comment:
it looks like with the links here/in the bbl column, we are no longer logging the `addressChangePortfolio` event--intentional?